### ISSUE TITLE
Can I use this with args on MacOS?

### ISSUE THREAD

**Issue by a-christie** [maidsafe/system_uri#52]
I managed to get this to work on MacOS. (Would be helpful to add a line to the readme stating that you have to bundle the app for this to work on MacOS.)
The app does open but if I provide an argument to the link, e.g. "scheme://ARG", is there a way for me to use that argument inside my app?

**Comment by hunterlester** [comment]
Yes, you are right about documentation needed for MacOS, thank you.

In our network browser, we obtain the passed argument either through Electron's `second-instance` [event](https://github.com/maidsafe/safe_browser/blob/dev/app/main.dev.ts#L119) or Electron's MacOS-specific `open-url` [event](https://github.com/hunterlester/safe-app-base/blob/master/main.js#L82).

What kind of application are you building and which language?

**Comment by a-christie** [comment]
Thanks for the quick response!
I was using Electron until recently but am now directly using your code in a compiled Rust app as I don‘t really need a GUI and therefore Electron is a lot of unnecessary dead weight to ship.

I‘m trying to make a custom protocol that allows me to show files and folders in the native explorer from within the browser (Reveal in Finder on MacOS and Reveal in Explorer on Windows). I use a jsonwebtoken to verify that only my application can launch that kind of link. 

My application works really well on Windows as on that platform when the application is launched via the URI Windows simply passes the whole URL as a argument to the app. However, this doesn‘t seem to work like that on MacOS.

Would you have any hints as to what I could try?

Thanks!
